src/map/lua/lua_baseentity.cpp

```diff
@@ -976,6 +976,13 @@ inline int32 CLuaBaseEntity::startEvent(lua_State *L)
         ((CCharEntity*)m_PBaseEntity)->StatusEffectContainer->DelStatusEffect(EFFECT_HEALING);
     }
 
+    CBattleEntity* PPet = ((CBattleEntity*)m_PBaseEntity)->PPet;
```

Comment:
Going forward, we're trying to remove C-Style casting for complex types.

Here's what something like this should look like and I'll explain what it's doing after (for those that care)

At the very top of `startEvent` we need something like this:
```
auto PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
if (!PChar)
{
    ShowError("CLuaBaseEntity::startEvent: Could not start event, Base Entity is not a Character Entity.\n");
    return 0;
}
```
This needs to happen before `int32 n = lua_gettop(L);`

We have a Debug Break at the top, making sure that this function is being invoked on a TYPE_PC entity. The dynamic cast will return a nullptr if m_PBaseEntity is anything other than a CCharEntity (e.g. CMobEntity) and will protect from people trying to call startEvent on mobs or NPCs. Instead we'll get an error and go about our day.

Furthermore, this new PChar pointer can be used throughout the rest of the function instead of continuously doing casts. Casts can be expensive.

Now your new code can become:
```
if (PChar->PPet)
{
    PChar->PPet->PAI->Disengage();
}
```
Likewise, throughout the rest of this function, anywhere you see `((CBattleEntity*)m_PBaseEntity)` can be replaced with our new PChar pointer.

Also note that this fix needs to be added to `startEventString`.

Not trying to pile up on you, just trying to get some of this code cleaned up while we can and provide good examples for the future.

Thanks!